ddtrace/ext/ci.py

```diff
@@ -41,28 +44,37 @@
 
 
 def _normalize_ref(name):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_TAGS.sub("", _RE_ORIGIN.sub("", _RE_REFS.sub("", name))) if name is not None else None
 
 
 def _filter_sensitive_info(url):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_URL.sub("\\1", url) if url is not None else None
 
 
 def tags(env=None):
+    # type: (Optional[MutableMapping[str, str]]) -> Dict[str, str]
+    """Extract and set tags from provider environ, as well as git metadata."""
     env = os.environ if env is None else env
-    tags = {}
+    tags = {}  # type: Dict[str, Optional[str]]
+    ci_tags = {}  # type: Dict[str, Optional[str]]
     for key, extract in PROVIDERS:
         if key in env:
-            tags = extract(env)
+            ci_tags = extract(env)
             break
 
+    git_info = git.extract_git_metadata()
+    # Tags collected from CI provider take precedence over extracted git metadata
+    tags.update(git_info)
+    tags.update(ci_tags)
```

Comment:
this behavior is different than before, no?

before I think if you had:

```python
ci_tags = {"a": 1, "b": None}
git_tags = {"a": 2, "b": 3}
```

the end result before would be:

```python
tags = {"a": 1, "b": 3}
```

where now with these changes we would end up with:

```python
tags = {"a": 1, "b": None}
```

Reply:
at least it doesn't match with the comment that states `tags` gets precedence. Also I just realised @Kyle-Verhoog made my same comment 6 days ago 🤦 .

Reply:
From what I understand, CI provider tags will either have a tag or not have it at all in almost all cases, i.e:
```python
ci_tags = {"a": 1}
git_tags = {"a": 2, "b": 3}
```
In the case that CI provider tags have any value, we always want to default to them rather than the Git extracted values. I'll double check in the weird case we have None values in the CI tags which tags to prioritize.

Reply:
it would be anything that is falsey, since before you had something like:

```python
tags.update({k: v for k, v in git_info.items() if not tags.get(k)})
```

I think either approach is fine, just want to make sure the code matches the comments matches the expectations

(maybe there are tests missing to confirm expected behavior? ;P)

Reply:
Haha it might be better to go back to the previous implementation. I'll add some tests to confirm that the code matches the comments 😆

Reply:
Double checked with the CI team - we want to prioritize the CI provider's git info over the git executable, but if the CI provider ever provides a None or empty string value, we want to overwrite the CI provider value by the git executable.